docs/Caching.md

```diff
@@ -0,0 +1,158 @@
+a request was made for a group event, the batched response would contain
+responses from the _group endpoint_ and the _event endpoint_. The app server
+could apply a `Cache-Control` header that matched the shortest cache
+time-to-live of those two responses, and the CDN could cache that combined
+response and deliver it to _every user who subsequently made the same batched
+request_. This cache behavior would not be useful if every batched request
+included an un-cacheable response (e.g. a private/logged-in response, like 
+`/members/self`), so it would have to be developed after the app was updated to
+cache individual API responses and only request stale data, as described below.
+
+3. **Redux middleware** - cache parsed API responses and filter `fetch`
+
+    The next level of caching would require changes to the query-based API proxy
+routine. Each individual query response would need to contain the cache control
+rules of the corresponding REST API response so that the browser could
+appropriately cache the responses _and avoid making the same query request_
+while the browser-cached response was available. A complintary caching stategy
+could be applied at the CDN layer - if the CDN was configured to cache responses
+from the `/mu_api` endpoint based on
+
+    Although a service worker might be able to provide a 'transparent' caching
+solution, the batched requests and responses would be awkward to parse in order
+to correctly refine them to include only stale data.
+
+    Instead, cache control rules should probably be handled by a more sophisticated 
+interaction between the Sync epic (middleware) and the Cache epic
+to allow `fetch` requests to be modified (or cancelled) when cached data was
+available for the requested queries. We would probably still want to use the
+Cache epic to cache everything so that the UI was responsive even when 'fresh'
+data was being fetched, but the cache control rules would allow the `fetch` to
+be fine-tuned to include only the data that was known to _require_ an update.
+
+## 2. Static assets
+
+Static assets generated by the build are cached in our CDN and in the browser.
+Because each filename includes a hash of the file contents, the file can have an
+infinite cache lifetime.
+
+### CDN
+
+(CDN cache invalidation policy?)
+
+### Browser
+
+All static assets are delivered to the browser with a `Cache-Control` header
+with a a 10 year expiration. For
+[browsers that do not support Service Workers](https://jakearchibald.github.io/isserviceworkerready/),
+normal browser cache rules apply - static assets will be cached as they are
+requested by the app, but not before. For browsers that _do_ support service
+workers, the platform service worker will 'pre-cache' all bundled assets
+asynchronously after the in-page assets load, and serve those assets from the
+browser cache for all future requests.
+
+When new assets are generated, a new service worker is generated. When a client
```

Comment:
how does the client know to download the new service worker? does it have a hashed filename like other built assets?